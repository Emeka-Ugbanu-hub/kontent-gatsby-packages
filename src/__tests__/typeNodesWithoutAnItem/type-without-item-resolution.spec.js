
const { KontentTestHttpService }
= require('@kentico/kontent-test-http-service-js');

const { sourceNodes } = require('../../gatsby-node');

const fakeTypeResponse =
require('./fakeTypesResponse.json');
const fakeItemsResponse =
require('./fakeItemsResponse.json');

describe(
  `Simple content type without item representation`,
  async () => {
    const fakeHttpServiceConfig = new Map();
    fakeHttpServiceConfig.set(
      /https:\/\/deliver.kontent.ai\/.*\/items/,
      {
        fakeResponseJson: fakeItemsResponse,
        throwError: false,
      });
    fakeHttpServiceConfig.set(
      /https:\/\/deliver.kontent.ai\/.*\/types/,
      {
        fakeResponseJson: fakeTypeResponse,
        throwError: false,
      });

    const dummyCreateNodeID = jest.fn();
    dummyCreateNodeID.mockImplementation((input) => `dummy-${input}`);

    const createNodeMock = jest.fn();
    const createTypesMock = jest.fn();

    const actions = {
      actions: {
        createNode: createNodeMock,
        createTypes: createTypesMock,
      },
      createNodeId: dummyCreateNodeID,
    };
    const deliveryClientConfig = {
      projectId: 'dummyProject',
      typeResolvers: [],
      httpService: new KontentTestHttpService(
        fakeHttpServiceConfig
      ),
    };

    const pluginConfiguration = {
      deliveryClientConfig,
      languageCodenames: ['default'],
    };

    it('passes with no error', async () => {
      await sourceNodes(actions, pluginConfiguration);

      const calls = createTypesMock.mock.calls;
      expect(calls).toMatchSnapshot();
    });
  });
