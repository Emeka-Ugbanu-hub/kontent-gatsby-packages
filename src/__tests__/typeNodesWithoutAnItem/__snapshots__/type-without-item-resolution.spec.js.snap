// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple content type without item representation passes with no error 1`] = `
Array [
  Array [
    "type KontentTypeWithoutitem implements Node { 
      photos: String ,title: String 
  }",
  ],
]
`;
